Author: Georg-Johann Lay
Description: Enhancement of a bugfix by Nils Carlson.
 Replaced a cc==0 check with proper Tcl_Eof() check.
Last-Modified: Fri, 14 Jul 2017 17:07:47 +0300
Bug: http://sourceforge.net/p/expect/patches/18/
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=799301
See-Also: https://gcc.gnu.org/ml/gcc/2017-06/msg00000.html

--- a/expect.c
+++ b/expect.c
@@ -1861,7 +1861,7 @@
 	/* try to read it */
 	cc = expIRead(interp,esPtr,timeout,tcl_set_flags);
 	
-	if (Tcl_Eof(esPtr->channel)) {
+	if (cc == 0 && Tcl_Eof(esPtr->channel)) {
 	    cc = EXP_EOF;
 	}
     } else if (cc == EXP_DATA_OLD) {
